### PR TITLE
V8: Remember editor language between sessions

### PR DESCRIPTION
**Pull request by kjac**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

### Description

When you have multiple languages it would be nice if the last selected "main language" was remembered between sessions.

This PR does just that:

persist-editor-language

The persistence is done clientside using a cookie (*UMB_MCULTURE*). While that obviously doesn't work between machines, I'm going to venture a bet that it'll solve the problem for 90-something percent of all editors. The selected language could also be stored serverside, but is it really worth it?

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/controllers/navigation.controller.js [hunk 1]
```diff
@@ -391,6 +399,8 @@ function NavigationController($scope, $rootScope, $location, $log, $q, $routePar
     $scope.selectLanguage = function (language) {
 
         $location.search("mculture", language.culture);
+        // add the selected culture to a cookie so the user will log back into the same culture later on
+        $cookies.put("UMB_MCULTURE", language.culture, {path: "/", expires: "2100-01-01"});
```
Comment (nul800sebastiaan) [inline src/Umbraco.Web.UI.Client/src/controllers/navigation.controller.js]: Could you use `max-age` instead and set it to 1 year? This is a tiny concern but some browser seems to be affected by the 2038 bug. Also, what if we're still using this site in the year 2101? 😉

Comment (kjac) [reply src/Umbraco.Web.UI.Client/src/controllers/navigation.controller.js]: Absolutely, I'll have a look at it 👍

Comment (kjac) [reply src/Umbraco.Web.UI.Client/src/controllers/navigation.controller.js]: @nul800sebastiaan done. Unfortunately $cookies doesn't support `max-age` (see https://docs.angularjs.org/api/ngCookies/provider/$cookiesProvider#defaults), but it works just as fine setting the cookie the good ol' JS way (`document.cookie = ...`).

**Comment by nul800sebastiaan** [comment]
Thanks @kjac I tried a few different scenarios (cookie, no cookie, default languages etc,) all good. 👍

**Comment by nul800sebastiaan** [comment]
Ah, I found an interesting edge case though @kjac - and a potential for a new PR.. lol

If I fiddle with the cookie and set it to `dk-DK` (which is not a language I've added to the settings) then the language dropdown above the content tree is empty. We should only switch to a language that we know exists, so it should've defaulted back to `en-US` in that case.

I guess this also becomes a problem when languages get deleted over time, so it would be nice to fix.

**Comment by kjac** [comment]
@nul800sebastiaan nice find 👍 I'll make sure to push a fix.

@bjarnef I can't test whether `$cookies` uses `expires` or `max-age`. All I can see is it only accepts `expires`, and Chromes developer console gives me no help. I'm happy to revert to using `$cookies` instead of `document.cookies` if @nul800sebastiaan is too.

**Comment by nul800sebastiaan** [comment]
I don't see the benefit of `$cookies` really but if it works then I'm happy :-)

**Comment by bjarnef** [comment]
@kjac I think the when setting `$cookies` expires it sets the `max-age` as I remember. We can of course also use traditional javascript,  just `$cookies` are used in most part of the Angular controllers etc. in Umbraco to handle cookies 🙂

**Comment by kjac** [comment]
Fixed in #4051

**Pull request merged**